impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -35,70 +27,19 @@
 import org.apache.maven.api.services.xml.XmlReaderRequest;
 import org.apache.maven.api.services.xml.XmlWriterException;
 import org.apache.maven.api.services.xml.XmlWriterRequest;
-import org.apache.maven.plugin.descriptor.io.PluginDescriptorStaxReader;
-import org.apache.maven.plugin.descriptor.io.PluginDescriptorStaxWriter;
-
-import static org.apache.maven.impl.ImplUtils.nonNull;
-import static org.apache.maven.impl.StaxLocation.getLocation;
-import static org.apache.maven.impl.StaxLocation.getMessage;
 
 @Named
 @Singleton
 public class DefaultPluginXmlFactory implements PluginXmlFactory {
+
     @Override
     public PluginDescriptor read(@Nonnull XmlReaderRequest request) throws XmlReaderException {
```

Comment:
```suggestion
    public PluginDescriptor createReadPlugin(@Nonnull XmlReaderRequest request) throws XmlReaderException {
```

Reply:
Then this name is kind of wrong. Why should a factory read?
Its producing goods therefore providing `createReadPlugin` or `createRead`

Reply:
But that makes no sense here. There's no creation of `read` objects.

Reply:
yes